RocketChat/Rocket.Chat
# [FIX] ASCII emojis for Marked Markdown

## Proposed changes

With Marked Markdown, emojis like `:)` aren't parsed correctly if there are no spaces or content before and after the emoji (`  :)  ` worked while `:)` didn't work). 
When comparing Original & Marked Markdown, I noticed that Marked Markdown had `<p>` HTML tags and Original hadn't. I added a function to remove those which solved the emoji transformation issue. 

When using Smartypants with Marked Markdown, emojis like :') weren't transformed either. This is due to Smartypants replacing ' to ’ (using `:\'(` transformed the emoji correctly in Marked Markdown while it didn't work in Original Markdown). I added a function to transform `’` back to `'` for emoji parsing.

## Issue(s)

Closes #11214 (for the global emoji transformation issue)

## How to test or reproduce

- Use Marked Markdown
- write a new message containing `:)`
- notice that the emoji isn't converted

## Screenshots
![image](https://user-images.githubusercontent.com/1096435/82122492-5679d380-9794-11ea-98e5-55d51b54d5fc.png)


## Types of changes




## Checklist



## Changelog

- Remove `<p>` & `</p>` while parsing emojis
- Transform `’` to `'` for emoji parsing (and back to `'` after the parsing)


## Further comments

I'm not sure removing the `<p>` & `</p>` HTML tags in the emoji parser is the best place to do it (?).
